linkerd/admin/src/main/resources/io/buoyant/linkerd/admin/js/router_summary.js

```diff
@@ -64,6 +64,7 @@ var RouterSummary = (function() {
     renderRouterSummary({ router: routerName }, summaryTemplate, routerName, $summaryEl);
 
     metricsCollector.registerListener(function(data) {
+        routers.update(data.general);
```

Comment:
Instead of routers being updated by one of many possible modules, could you pass metricsCollector to routers and have it register itself for updates? It's annoying, but in the end cleaner